projects/storefrontapp-e2e-cypress/cypress/integration/regression/my-account/my-coupons.e2e-spec.ts

```diff
@@ -0,0 +1,44 @@
+import * as myCoupons from '../../../helpers/my-coupons';
+
+describe('My coupons test for anonymous user', () => {
+  before(() => {
+    cy.window().then(win => win.sessionStorage.clear());
+  });
+
+  myCoupons.verifyMyCouponsAsAnonymous();
+
+  describe('claim coupon test for anonymous user', () => {
+    beforeEach(() => {
+      myCoupons.registerUser();
+    });
+    myCoupons.verifyClaimCouponSuccessAsAnonymous(myCoupons.validCouponCode);
+    myCoupons.verifyClaimCouponFailedAsAnonymous(myCoupons.invalidCouponCode);
+  });
+});
+
+describe('My coupons test for logged in user', () => {
+  before(() => {
+    cy.window().then(win => win.sessionStorage.clear());
+    cy.requireLoggedIn();
+    cy.reload();
+    cy.visit('/');
+    cy.selectUserMenuOption({
+      option: 'My Coupons',
+    });
+  });
+  myCoupons.verifyMyCoupons();
+});
+
+describe('My coupons test for pagination and sort', () => {
+  before(() => {
+    cy.window().then(win => win.sessionStorage.clear());
+    cy.login(myCoupons.testUser, myCoupons.testPassword);
+    cy.reload();
+    cy.visit('/');
+    cy.selectUserMenuOption({
+      option: 'My Coupons',
+    });
+  });
+
+  myCoupons.verifyPagingAndSorting();
+});
```

Comment:
Maybe a test case that contains [claim->add to cart->apply a coupon via auto-complate function] is necessary.

Reply:
added in cart coupon e2e case, now move to my coupons